batch/test/test_batch.py

```diff
@@ -1486,6 +1486,31 @@ def test_job_private_instance_cancel(client: BatchClient):
     assert status['state'] == 'Cancelled', str((status, b.debug_info()))
 
 
+def test_always_run_job_private_instance_cancel(client: BatchClient):
+    bb = create_batch(client)
+    resources = {'machine_type': smallest_machine_type()}
+    j = bb.create_job(DOCKER_ROOT_IMAGE, ['true'], resources=resources, always_run=True)
+    b = bb.submit()
+
+    tries = 0
+    start = time.time()
+    while True:
+        status = j.status()
+        if status['state'] == 'Creating':
+            break
+        now = time.time()
```

Comment:
Why do you need to poll here?

Reply:
I copy pasted this test from `test_job_private_instance_cancel`. My guess is that we don't want to cancel until the job is actually creating (ergo we've actually created an instance).

Reply:
It's always run though. Cancel shouldn't do anything.

Reply:
Can you get rid of the polling? It's not necessary.